h2o-core/src/main/java/water/Paxos.java

```diff
@@ -77,6 +78,11 @@ static synchronized int doHeartbeat( H2ONode h2o ) {
         // themselves via heartbeat to the client
         H2O.addNodeToFlatfile(h2o);
       }
+    }else{
+      // We are operating in the multicast mode and heard from the client, so we need to report the client on this node
```

Comment:
Moved from UDPHeartbeat?

Reply:
Yup, exactly. It can be located here as this code is called always from `UDPHeartBeat`. I moved it so we can have all the cloud/client handling together at single place and not scattered around